### PR TITLE
Removed Deprecated Pulse Call Instruction

### PR DESCRIPTION
**Pull request by sbrandhsn**
### Summary

Removed pulse call instruction deprecated in #10247 and #10103

### Details and comments

I was a bit unsure about removing all test cases that use `pulse.Call `since it seemed that `pulse.Call `sometimes only provides reference values for some of these, but proceeded after reading the discussion in #10103. I hope that was okay.

But just to be sure, it would be great if you (maybe @nkanazawa1989 ? :-) ) can check that all removed test cases are indeed included in https://github.com/Qiskit/qiskit/blob/main/test/python/pulse/test_builder.py or https://github.com/Qiskit/qiskit/blob/main/test/python/pulse/test_builder_v2.py.
I'm especially sceptical that everything in `TestParameterGetter` and `TestParameterSetter` could be removed -- there, `pulse.Call` was used to create some reference values.

### CODE REVIEW

File: test/python/pulse/test_parameter_manager.py [hunk 1]
```diff
@@ -14,491 +14,12 @@
...
-
 from qiskit import pulse
 from qiskit.circuit import Parameter
 from qiskit.pulse.exceptions import PulseError, UnassignedDurationError
-from qiskit.pulse.parameter_manager import ParameterGetter, ParameterSetter
-from qiskit.pulse.transforms import AlignEquispaced, AlignLeft, inline_subroutines
 from qiskit.test import QiskitTestCase
 
 
-class ParameterTestBase(QiskitTestCase):
```
Comment (ElePT) [inline test/python/pulse/test_parameter_manager.py]: I haven't taken the time to analyze the test in depth, but would it make sense to keep it and only delete/replace the `schedule.Call` line?

**Review by ElePT — COMMENTED** [review]
The removal procedure looks well executed :), I think I don't have enough pulse expertise to judge if any additional unit test is necessary to compensate for the removal of `Call`.

File: test/python/pulse/test_parameter_manager.py (hunk 1 above)
Comment (sbrandhsn) [reply test/python/pulse/test_parameter_manager.py]: Thanks for taking a look! :) Yes, there are a couple of test cases that could qualify for removal or rephrasing. The discussion in issue #10103 lead me to believe there are already alternatives for test cases using `Pulse.Call`, allowing for current `Pulse.Call` test cases to be removed... Do you suggest to rephrase all test cases using `builder.call` instead of `Pulse.Call`?

File: qiskit/pulse/library/symbolic_pulses.py [hunk 2]
```diff
@@ -598,19 +597,6 @@ class ScalableSymbolicPulse(SymbolicPulse):
     :math:'\text{amp}\times\exp\left(i\times\text{angle}\right)' is compared.
     """
 
-    @deprecate_arg(
```
Comment (nkanazawa1989) [inline qiskit/pulse/library/symbolic_pulses.py]: Removal of deprecated code in this file is done by https://github.com/Qiskit/qiskit/pull/11403, please revert this change not to conflict.

File: qiskit/qpy/binary_io/schedules.py [hunk 3]
```diff
@@ -137,33 +136,17 @@ def _read_symbolic_pulse(file_obj, version):
         vectors={},
     )
 
-    # In the transition to Qiskit Terra 0.23 (QPY version 6), the representation of library pulses
```
Comment (nkanazawa1989) [inline qiskit/qpy/binary_io/schedules.py]: same here

File: releasenotes/notes/deprecate-pulse-instruction-call-52dca0dd26e1c768.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - |
+    Removed the deprecated class :class:`qiskit.pulse.instructions.Call`. Use function
+    :func:`qiskit.pulse.builder.call` instead.
```
Comment (nkanazawa1989) [inline releasenotes/notes/deprecate-pulse-instruction-call-52dca0dd26e1c768.yaml]:
```suggestion
    Removed the deprecated class :class:`qiskit.pulse.instructions.Call`. 
    No alternative pulse instruction is provided.
```

The `Call` instruction used to be a wrapper of `Schedule` so that `ScheduleBlock` can call `Schedule` program (device calibration) in the builder context as a subroutine. This was needed to handle `Schedule` like an instruction, but now `Schedule` injected into `ScheduleBlock` context is implicitly converted into `ScheduleBlock` and flattened. So this instruction is just no longer necessary.

File: test/python/pulse/test_builder.py [hunk 5]
```diff
@@ -811,31 +811,6 @@ def get_sched(qubit_idx: [int], backend):
...
-        d0 = pulse.DriveChannel(0)
-        d1 = pulse.DriveChannel(1)
-
-        reference = pulse.Schedule()
-        reference = reference.insert(10, instructions.Delay(10, d0))
-        reference += instructions.Delay(20, d1)
-
-        ref_sched = pulse.Schedule()
-        with self.assertWarns(DeprecationWarning):
-            ref_sched += pulse.instructions.Call(reference)
```
Comment (nkanazawa1989) [inline test/python/pulse/test_builder.py]: I think you can just add `reference` directly here instead of casting it into Call. This test is against builder call command, so should not be removed.

File: test/python/pulse/test_parameter_manager.py (hunk 1 above)
Comment (nkanazawa1989) [reply test/python/pulse/test_parameter_manager.py]: You shouldn't remove these tests. This file covers the test cases for schedule parameter assignment.

File: test/python/pulse/test_parameter_manager.py [hunk 6]
```diff
@@ -14,491 +14,12 @@
...
-        self.reg1 = Parameter("m1")
-
-        self.context_dur = Parameter("context_dur")
-
-        # schedule under test
-        subroutine = pulse.ScheduleBlock(alignment_context=AlignLeft())
-        subroutine += pulse.ShiftPhase(self.phi1, self.d1)
-        subroutine += pulse.Play(self.parametric_waveform1, self.d1)
-
-        sched = pulse.Schedule()
```
Comment (nkanazawa1989) [inline test/python/pulse/test_parameter_manager.py:81]: You just need to remove this `sched` and directly insert 
```python
sched += pulse.ShiftPhase(self.phi3, self.d3)
```
into `long_schedule`, namely, in line 92
```python
long_schedule += pulse.ShiftPhase(self.phi3, self.d3)
```

This test checks if parameters can be assigned to `Schedule` wrapped by the `Call` instruction, as they have a bit tricky parameter management. Since `Call` instruction is removed with this PR, you don't need to test this. However, rest of program should be tested as before.

File: test/python/pulse/test_parameter_manager.py [hunk 7]
```diff
@@ -14,491 +14,12 @@
...
-        test_obj = pulse.ShiftPhase(self.phi1 + self.phi2, pulse.DriveChannel(0))
-
-        visitor = ParameterGetter()
-        visitor.visit(test_obj)
-
-        ref_params = {self.phi1, self.phi2}
-
-        self.assertSetEqual(visitor.parameters, ref_params)
-
-    def test_get_parameter_from_call(self):
```
Comment (nkanazawa1989) [inline test/python/pulse/test_parameter_manager.py:152]: You can remove this.

File: test/python/pulse/test_parameter_manager.py [hunk 8]
```diff
@@ -14,491 +14,12 @@
...
-        value_dict = {self.phi1: 0.123, self.phi2: 0.456}
-
-        visitor = ParameterSetter(param_map=value_dict)
-        assigned = visitor.visit(test_obj)
-
-        ref_obj = pulse.ShiftPhase(0.579, pulse.DriveChannel(0))
-
-        self.assertEqual(assigned, ref_obj)
-
-    def test_set_parameter_to_call(self):
```
Comment (nkanazawa1989) [inline test/python/pulse/test_parameter_manager.py:258]: You can remove this.

File: test/python/pulse/test_parameter_manager.py [hunk 9]
```diff
@@ -14,491 +14,12 @@
...
-        value_dict = {self.context_dur: 1000, self.dur1: 100}
-
-        visitor = ParameterSetter(param_map=value_dict)
-        assigned = visitor.visit(test_obj)
-
-        ref_obj = AlignEquispaced(duration=1100)
-
-        self.assertEqual(assigned, ref_obj)
-
-    def test_nested_assignment_partial_bind(self):
```
Comment (nkanazawa1989) [inline test/python/pulse/test_parameter_manager.py]: Please keep this test. You can just remove dependency on Call and add subroutine directly here because subroutine is ScheduleBlock.

File: test/python/pulse/test_parameter_manager.py [hunk 10]
```diff
@@ -14,491 +14,12 @@
...
-                pulse.Play(
-                    pulse.Gaussian(duration=100, amp=0.5, sigma=sig, angle=0.0),
-                    pulse.DriveChannel(0),
-                ),
-                inplace=True,
-            )
-            binded_sched = test_sched.assign_parameters({sig: -1.0}, inplace=False)
-            self.assertLess(binded_sched.instructions[0][1].pulse.sigma, 0)
-
-    def test_set_parameter_to_complex_schedule(self):
```
Comment (nkanazawa1989) [inline test/python/pulse/test_parameter_manager.py]: Please keep this as well (you just need to remove dependency on Call)

File: test/python/pulse/test_pulse_lib.py [hunk 11]
```diff
@@ -147,32 +147,6 @@ def test_construction(self):
         Sech(duration=50, amp=0.5, sigma=10, zero_ends=False)
         SechDeriv(duration=50, amp=0.5, sigma=10)
 
-    # This test should be removed once deprecation of complex amp is completed.
```
Comment (nkanazawa1989) [inline test/python/pulse/test_pulse_lib.py]: This file can be also reverted because this is done by https://github.com/Qiskit/qiskit/pull/11403

**Review by nkanazawa1989 — CHANGES_REQUESTED** [review]
Thanks @sbrandhsn for taking care of this. I really appreciate. My comment is largely about these two:

- You can revert changes to symbolic pulse, since this is done by https://github.com/Qiskit/qiskit/pull/11403
- Please do NOT delete the entire parameter assignment test `test.python.pulse.test_parameter_manager`. You can just remove dependency on `Call` from the reference object.

**Comment by sbrandhsn** [comment]
Arigatou gozaimashita, @nkanazawa1989 ! :-) I was not familiar with the pulse part of qiskit, thanks for clarifying the deprecation of Pulse.Call!  

This PR also removes:
- test_parameters_from_subroutine
- test_assign_parameter_to_subroutine
- test_assign_parameter_to_subroutine_parameter

**Review by nkanazawa1989 — APPROVED** [review]
@sbrandhsn Thanks for quick update!
